src/hooks/useIsIntegrationInstalled.ts

```diff
@@ -5,10 +5,13 @@ import { useIntegrationList } from '../context/IntegrationListContextProvider';
 import { useProject } from '../context/ProjectContextProvider';
 import { api, Installation, Integration } from '../services/api';
 
-interface UseIsIntegrationInstalledResult {
-  isLoaded: boolean;
-  isIntegrationInstalled: boolean | null;
-}
+type UseIsIntegrationInstalledResult = {
+  isLoaded: false;
+  isIntegrationInstalled: null;
+} | {
+  isLoaded: true;
+  isIntegrationInstalled: boolean;
+};
```

Comment:
The previous type was incorrect. There are two possible states for `UseIsIntegrationInstalledResult`, which is provided here. Checking `isLoaded` is enough to know whether `isIntegrationInstalled` is `null` or not.

Reply:
Possibly not incorrect, but the change looks more clear and precise!